server/errors.go

```diff
@@ -171,46 +171,10 @@ var (
 	ErrMalformedSubject = errors.New("malformed subject")
 
 	// ErrSubscribePermissionViolation is returned when processing of a subscription fails due to permissions.
-	ErrSubscribePermissionViolation = errors.New("subscribe permission viloation")
+	ErrSubscribePermissionViolation = errors.New("subscribe permission violation")
 
 	// ErrNoTransforms signals no subject transforms are available to map this subject.
 	ErrNoTransforms = errors.New("no matching transforms available")
-
-	// ErrJetStreamNotEnabled is returned when JetStream is not enabled.
-	ErrJetStreamNotEnabled = errors.New("jetstream not enabled")
```

Comment:
As we discussed once, should we leave those but with a doc section that says that they are "deprecated"? I am thinking of use-case where users embed the NATS Server. I assume that internally they will no longer be used or updated.

Reply:
Yeah its a difficult one, we could say `ErrJetStreamNotEnabled = ApiErrors[JSWhatever]` but I am not 100% sure that will solve the problem in all cases tbh

Reply:
Actually it should solve it for these - as they are essentially static, the ones where it wont solve it are the ones where we put in a different description via printf.  So it might be fine

Reply:
We should leave, no harm imo..

Reply:
I think the point was that we would not use them anymore, but we may want to keep them as to possibly not break users that embed NATS Server and somehow were checking those error codes?

Reply:
Yeah, I ppreciate why we would want to keep them - however, keeping them would not solve the problem completely.

For some we can keep them via an alias asignment but any of them that were dynamic would not be possible (the ones with tokens in them):

```
ErrJetStreamNotEnabled = ApiErrors[Foo]
```

Non of these had tokens of course so I need to look, might have to adjust `NewT()` and related functions to instead of a new instance return the existing pointer that is in `ApiErrors`, thats the only way it could possibly work.  This though is risky someone could adjust the "source" of these errors and bad things will happen, as there's no way to really make them read only as such

Reply:
Yes, it silly of me to say to keep them not to "break" their code (because say `ErrJetStreamNotEnabled` is no longer defined), while it would break it in worse way by having the check for say `ErrJetStreamNotEnabled` be false because of the new errors returned.
So I would be ok if we remove in 2.3.0, but then if we have to release a 2.2.7 before that release, then we would have to create and maintain a `branch_2_2_x` to release from, which I would rather not have to do..

Reply:
Old costants were restored in https://github.com/nats-io/nats-server/pull/2168/commits/ee9d10f40b8f16aacddd57f8142df7004422a532 with appropriate comments warning people etc

I think this one is good to go now